docs/FAQs.md

```diff
@@ -13,6 +13,131 @@ After you set up a repository, you can substitute in any tools you'd like.
+     Now we need to update the `build` script in our `package.json`:
+
+   ```diff
+   -"build": "tsup",
+   +"build": "ncc build src/index.ts -o dist --license licenses.txt",
+   ```
+
+   - Our build now emits to the `dist` directory; so we'll want to avoid linting that directory by adding the following to `.eslintignore` and our `.prettierignore`:
+
+   ```diff
+   +dist
+   ```
+
+   - Rather than having to remember to compile each time, we'll update our precommit hook in `.husky/precommit` to build for us on each commit:
+
+   ```diff
+   +pnpm run build
+   +git add dist
+   npx lint-staged
+   ```
+
+   - on the off chance someone isn't running husky, we'll add `.github/workflows/check-dist.yml` ([based on this](https://github.com/actions/typescript-action/blob/main/.github/workflows/check-dist.yml)) to ensure `dist` is up to date:
+
+   ```yml
+   # In TypeScript actions, `dist/` is a special directory. When you reference
+   # an action with the `uses:` property, `dist/index.js` is the code that will be
+   # run. For this project, the `dist/index.js` file is transpiled from other
+   # source files. This workflow ensures the `dist/index.js` file contains the
+   # expected transpiled code.
+   #
+   # If this workflow is run from a feature branch, it will act as an additional CI
+   # check and fail if the checked-in `dist/` directory does not match what is
+   # expected from the build.
+   jobs:
+     check-dist:
+       name: check dist
+       runs-on: ubuntu-latest
+
+       steps:
+         - uses: actions/checkout@v4
+         - uses: ./.github/actions/prepare
+
+         - id: build
+           name: Build dist directory
+           run: pnpm run build
+
+         # This will fail the workflow if the PR wasn't created by Dependabot.
+         - id: diff
+           name: Compare directories
+           run: |
+             if [ "$(git diff --ignore-space-at-eol --text dist/index.js | wc -l)" -gt "0" ]; then
+               echo "Detected uncommitted changes after build."
+               echo "You may need to run 'pnpm run build' locally and commit the changes."
+               echo ""
+               echo "See diff below:"
+               echo ""
+               git diff --ignore-space-at-eol --text dist/index.js
+               echo ""
+               # say this again in case the diff is long
+               echo "You may need to run 'pnpm run build' locally and commit the changes."
+               echo ""
+               exit 1
+             fi
+
+         # If `dist/` was different than expected, and this was not a Dependabot
+         # PR, upload the expected version as a workflow artifact.
+         - id: upload
+           if: ${{ failure() && steps.diff.outcome == 'failure' }}
+           name: Upload artifact
+           uses: actions/upload-artifact@v4
+           with:
+             name: dist
+             path: dist/
```

Comment:
[Question] Why? Who needs this artifact?

```suggestion
```

Reply:
Good question - I don't know.  I can remove it?  It came from where I borrowed it in the first place - quite possibly not required

Reply:
Yeah I say let's remove it & discover any gaps the hard way.

Reply:
Yeah agreed - I'll remove. Can probably get to this tonight if all goes smoothly